sympy/solvers/solveset.py

```diff
@@ -998,6 +1006,252 @@ def _solveset(f, symbol, domain, _check=False):
 
     return result
 
+def _is_modular(f, symbol):
+    """
+    Helper function to check below mentioned types of modular equations.
+    ``A - Mod(B, C) = 0``
+
+    A -> This can or cannot be a function of symbol.
+    B -> This is surely a function of symbol.
+    C -> It is an integer.
+
+    Parameters
+    ==========
+
+    f : Expr
+        The equation to be checked.
+
+    symbol : Symbol
+        The concerned variable for which the equation is to be checked.
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, exp, Mod
+    >>> from sympy.solvers.solveset import _is_modular as check
+    >>> x, y = symbols('x y')
+    >>> check(Mod(x, 3) - 1, x)
+    True
+    >>> check(Mod(x, 3) - 1, y)
+    False
+    >>> check(Mod(x, 3)**2 - 5, x)
+    False
+    >>> check(Mod(x, 3)**2 - y, x)
+    False
+    >>> check(exp(Mod(x, 3)) - 1, x)
+    False
+    >>> check(Mod(3, y) - 1, y)
+    False
+    """
+
+    if not f.has(Mod):
+        return False
+
+    # extract modterms from f.
+    modterms = list(f.atoms(Mod))
+
+    return (len(modterms) == 1 and # only one Mod should be present
+            modterms[0].args[0].has(symbol) and # B-> function of symbol
+            modterms[0].args[1].is_integer and # C-> to be an integer.
+            any(isinstance(term, Mod)
+            for term in list(_term_factors(f))) # free from other funcs
+            )
+
+
+def _invert_modular(modterm, rhs, n, symbol):
+    """
+    Helper function to invert modular equation.
+    ``Mod(a, m) - rhs = 0``
+
+    Generally it is inverted as (a, ImageSet(Lambda(n, m*n + rhs), S.Integers)).
+    More simplified form will be returned if possible.
+
+    If it is not invertible than modterm, rhs is returned.
+
+    Parameters
+    ==========
+
+    modterm, rhs : Expr
+        The modular equation to be inverted, ``modterm - rhs = 0``
+
+    symbol : Symbol
+        The variable in the equation to be inverted.
+
+    n : Dummy
```

Comment:
Why is this variable passed into the function instead of creating one inside the function. Also, since you're defining it right above the call of this function in `_solve_modular` [here](https://github.com/sympy/sympy/pull/16976/files#diff-eec0422923e8f100745c015cd8fdd6cfR1230).

Reply:
Because `_invert_modular` is a recursive function and if I create variable in `_invert_modular` then due to its recursive nature many useless instances will be created of n. So I defined it in `_solve_modular`